src/pages/Entries/ScrollableCalendar.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from "react";
+import { useMemo, useRef } from "react";
+import { useInfiniteQuery } from "react-query";
+import HappinessCard from "./HappinessCard";
+import InfiniteScroll from "react-infinite-scroll-component";
+import { useApi } from "../../contexts/ApiProvider";
+import Spinner from "../../components/Spinner";
+import { Happiness } from "../../data/models/Happiness";
+import { formatDate } from "../../utils";
+
+// type of the paginated happiness data structure used for infinite scroll
+interface HappinessPagination {
+  data: Happiness[];
+  page: number;
+}
+
+export default function ScrollableCalendar({
+  selectedEntry,
+  setSelectedEntry,
+}: {
+  selectedEntry: Happiness | undefined;
```

Comment:
When would `selectedEntry` be undefined?

Reply:
during the initial/loading state?